### PR TITLE
Plugin injection changes

### PR DESCRIPTION
**Pull request by kvanhijf**
Bugfix when no fragment cache and local block cache is used
Bugfix in datalist when paging is used in combination with custom made DataList

WIP!!!!!!

### CODE REVIEW

File: ovs/dal/tests/test_basic.py [hunk 1]
```diff
@@ -1773,3 +1773,40 @@ def _uses_indexes(datalist_object):
         self.assertEqual(len(dlist), 1)
         self.assertTrue(Basic._called)
         self.assertEqual(dlist.from_index, 'full')
+
+    def test_clone(self):
+        """
+        Validates whether the clone function works correct
```
Comment (JeffreyDevloo) [inline ovs/dal/tests/test_basic.py]: correctly*

File: webapps/frontend/app/widgets/pager/viewmodel.js [hunk 2]
```diff
@@ -212,7 +212,15 @@ define([
...
             }
             if (self.external()) {
-                self.loadData(options);
+                self.pageLoading(true);
+                var chainDeferred = $.Deferred(), chainPromise = chainDeferred.promise();
+                chainDeferred.resolve();
+                chainPromise.then(function() {
+                    return self.loadData(options);
+                })
+                    .always(function() {
```
Comment (JeffreyDevloo) [inline webapps/frontend/app/widgets/pager/viewmodel.js]: Indenting

File: ovs/dal/tests/test_basic.py [hunk 3]
```diff
@@ -1773,3 +1773,40 @@ def _uses_indexes(datalist_object):
...
+        """
+        machine1 = TestMachine()
+        machine1.name = 'test_machine1'
+        machine1.save()
+        machine2 = machine1.clone()
+        self.assertEqual(machine1.name, machine2.name)
+        self.assertEqual(machine1.guid, machine2.guid)
+        self.assertEqual(machine1._datastore_wins, machine2._datastore_wins)
+        self.assertEqual(machine1, machine2)
```
Comment (JeffreyDevloo) [inline ovs/dal/tests/test_basic.py]: We should test that the _data property is also identical in this case

**Review by JeffreyDevloo — CHANGES_REQUESTED** [review]

File: ovs/dal/tests/test_basic.py [hunk 4]
```diff
@@ -1785,6 +1785,7 @@ def test_clone(self):
         self.assertEqual(machine1.name, machine2.name)
         self.assertEqual(machine1.guid, machine2.guid)
         self.assertEqual(machine1._datastore_wins, machine2._datastore_wins)
+        self.assertEqual(machine1._data, machine2._data)
```
Comment (khenderick) [inline ovs/dal/tests/test_basic.py]: Use `assertDictEqual` here

**Review by khenderick — CHANGES_REQUESTED** [review]

**Review by JeffreyDevloo — APPROVED** [review]

**Review by khenderick — APPROVED** [review]

**Pull request merged**